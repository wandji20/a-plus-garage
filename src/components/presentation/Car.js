import React, { Fragment, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { CircularProgressbar } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import Part from './Part';
import Button from './Button';
import computeDisplayDetails from '../../helpers/computeDisplayInfo';

const Car = (props) => {
  const {
    car, parts,
  } = props;

  const {
    id, make, power, fuel,
  } = car;

  const carParts = parts.filter((part) => (part.car_id === id));

  useEffect(() => {

  }, [carParts]);

  const { allPartsInfo, overall, condition } = computeDisplayDetails(carParts);

  return (
    <>
      <div
        className="container remove-padding d-flex flex-column justify-content-center align-items-center py-3 bg-light"
      >
        <article className="d-flex flex-column justify-content-center align-items-center">
          <h3>
            <span>Make :</span>
            <span>
              {' '}
              {make}
            </span>
          </h3>
          <p>
            <span>Horse Power :</span>
            <span>
              {' '}
              {power}
            </span>
          </p>
          <p>
            <span>Average Weekly consumption per gallon :</span>
            <span>
              {' '}
              {fuel}
            </span>
          </p>
          <figure className="col-6">
            <CircularProgressbar
              value={overall}
              text={`${overall}%`}
            />
          </figure>
          <h3 className="d-flex flex-column align-items-center">
            <span>{condition.status}</span>
            <span>
              {' '}
              Working Condition
            </span>
          </h3>
        </article>
      </div>
      <div className="bg-light container-fluid d-flex-column align-items-start all-parts">
        {
          allPartsInfo
          && (
            allPartsInfo.map((part) => <Part part={part} key={part.id} />)
          )
        }
      </div>
      <div className="row py-2 justify-content-center">
        {
        (carParts && carParts.length > 0)
        && <Button id={id} />
      }

      </div>
    </>
  );
};

const mapStateToProps = (state) => ({
  parts: state.carsReducer.parts,
});

Car.propTypes = {
  car: PropTypes.objectOf(PropTypes.any).isRequired,
  parts: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default connect(mapStateToProps)(Car);
