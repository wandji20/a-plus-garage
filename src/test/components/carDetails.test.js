import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import Car from '../../components/presentation/Car';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Car id={1} />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
