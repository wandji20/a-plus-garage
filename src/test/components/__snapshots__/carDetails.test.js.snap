// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="container remove-padding d-flex flex-column justify-content-center align-items-center py-3 bg-light"
  >
    <article
      className="d-flex flex-column justify-content-center align-items-center"
    >
      <h3>
        <span>
          Make :
        </span>
        <span>
           
        </span>
      </h3>
      <p>
        <span>
          Horse Power :
        </span>
        <span>
           
        </span>
      </p>
      <p>
        <span>
          Average Weekly consumption per gallon :
        </span>
        <span>
           
        </span>
      </p>
      <figure
        className="col-6"
      >
        <svg
          className="CircularProgressbar "
          data-test-id="CircularProgressbar"
          style={Object {}}
          viewBox="0 0 100 100"
        >
          <path
            className="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "0px",
              }
            }
          />
          <path
            className="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "289.02652413026095px",
              }
            }
          />
          <text
            className="CircularProgressbar-text"
            style={Object {}}
            x={50}
            y={50}
          >
            0%
          </text>
        </svg>
      </figure>
      <h3
        className="d-flex flex-column align-items-center"
      >
        <span>
          
        </span>
        <span>
           
          Working Condition
        </span>
      </h3>
    </article>
  </div>,
  <div
    className="bg-light container-fluid d-flex-column align-items-start all-parts"
  />,
  <div
    className="row py-2 justify-content-center"
  />,
]
`;
